plutus-tx/src/PlutusTx/List.hs

```diff
@@ -244,14 +246,24 @@ xs0 !! n0 = go n0 xs0
             then x
             else go (Builtins.subtractInteger n 1) xs
 
-{-# INLINABLE reverse #-}
--- | Plutus Tx version of 'Data.List.reverse'.
-reverse :: [a] -> [a]
-reverse l = rev l []
-  where
+{-# INLINABLE appendR #-}
+-- | Cons each element of the first list to the second one in reverse order (i.e. the last element
+-- of the first list is the head of the result).
+--
+-- > appendR xs ys === reverse xs ++ ys
+--
+-- >>> appendR "abc" "de"
+-- "cbade"
+appendR :: forall a. [a] -> [a] -> [a]
+appendR = rev where
+    rev :: [a] -> [a] -> [a]
```

Comment:
Just replicating the old logic to keep the diff minimal. We could've recursed in `appendR` directly, but that would be more costly due to that type-level `a` argument becoming a bunch of `delay` and `force` on the UPLC side (which is really annoying).